### PR TITLE
Switched toolbox IDs to signed.

### PR DESCRIPTION
**Pull request by c-jo**

### CODE REVIEW

**Comment by laurenrad** [comment]
It's rather late/early here so I'm not sure if I'll be able to look into it right away, but this doesn't run at all, particularly as the new ObjectID / ComponentID classes aren't usable in place of ints.

**Comment by c-jo** [comment]
> It's rather late/early here so I'm not sure if I'll be able to look into it right away, but this doesn't run at all, particularly as the new ObjectID / ComponentID classes aren't usable in place of ints.

It was a bit of a WIP MR tbh. I've reverted them back to just ctypes.c_int32 now.

**Comment by gerph** [comment]
I guess the thing that matters is whether you can pass in a component number of 0xFFFFFFFF - as I recall that's the documented case for component ids.

> An object 'component' defines one of a set of distinct parts which make up an object; for example a menu entry is a component of a Menu object, and a gadget (see later) is a component of a Window object. A component is allocated a component id by which to identify it uniquely within its containing object; this component id is chosen by the client application when the component is created. For menus it can have a value in the range 0 to 0xFFFFFFFD, and for windows a value in the range 0 to 0x7FFFFF. All higher component ids are reserved for internal Toolbox use. A component id of 0xFFFFFFFF is used to indicate 'no component'.

This means that you have to allow the component id to be any number from 0 to 0xFFFFFFFD and the 0xFFFFFFFF at the very least. If that remains true for the change then it's ok.

The ObjectId is also an unsigned number, with no interpretation being placed upon it:

> At run-time, each object is identified by an object id which is allocated when the object is created. An object id is a 32-bit integer, which should not be interpreted by the client application. An object id of 0 is used to indicate 'no object'.

The only value that is to be interpreted is the value 0. So treating it as negative is probably fine, as nobody could ever interpret the value.

**Comment by laurenrad** [comment]
I concur, also now that the issue with those classes is out of the way, there is still something not lining up somewhere but it's Thanksgiving here in the States and I won't be able to take a look until later.

File: riscos_toolbox/gadgets/scrolllist.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def delete_items(self, start, end):
 
     def get_selected(self, offset=-1):
         return swi.swi('Toolbox_ObjectMiscOp', 'IiIii;i',
-                       0, self.window.id, 16416, Scrollist.GetSelected, self.id, offset)
+                       0, self.window.id, 16416, ScrollList.GetSelected, self.id, offset)
```
Comment (laurenrad) [inline riscos_toolbox/gadgets/scrolllist.py]: There's a duplicate argument in here, the old literal 16416 should have been removed when the ScrollList.GetSelected constant was added.

File: riscos_toolbox/gadgets/scrolllist.py
Comment (laurenrad) [inline riscos_toolbox/gadgets/scrolllist.py:1]: All of these method constants should be offsets from the base _type, so GetState should be:
`GetState = _type + 0` etc. Also, I noticed the state property isn't using these new named constants.

**Review by laurenrad — CHANGES_REQUESTED** [review]
I think I've figured out most of the trouble I was having earlier when I was trying to launch RLook, and it looks like it had to do with the ScrollList module.

File: riscos_toolbox/gadgets/scrolllist.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def delete_items(self, start, end):
 
     def get_selected(self, offset=-1):
         return swi.swi('Toolbox_ObjectMiscOp', 'IiIii;i',
-                       0, self.window.id, 16416, ScrollList.GetSelected, self.id, offset)
+                       0, self.window.id  ScrollList.GetSelected, self.id, offset)
```
Comment (laurenrad) [inline riscos_toolbox/gadgets/scrolllist.py]: Missing comma?

**Comment by c-jo** [comment]
Where's the facepalm-emoji ... :)
   
  
    
   
  
   Thanks
   
  
    
   
   
   
    On 24/11/2023 10:26 GMT Lauren Rad ***@***.***> wrote:
    
   
     
    
   
     
    
   
    @laurenrad commented on this pull request.
    
   
     
    
   
     
    
   
     
    
    
    
     @@ -45,7 +45,7 @@ def delete_items(self, start, end):
     
    
   
     
    
   
    def get_selected(self, offset=-1):
    
   
    return swi.swi('Toolbox_ObjectMiscOp', 'IiIii;i',
    
   
    - 0, self.window.id, 16416, ScrollList.GetSelected, self.id, offset)
    
   
    + 0, self.window.id ScrollList.GetSelected, self.id, offset)
    
   
     
    
   
    Missing comma?
    
   
     
    
   
    --
    
   
    Reply to this email directly or view it on GitHub:
    
   
    https://github.com/c-jo/riscos-toolbox/pull/76#pullrequestreview-1747700643

**Comment by c-jo** [comment]
Given the issues with IDs, I'm going to revert them to signed for now (so they don't change from v1.0) and look into this for a later release.

**Pull request merged**